[WIP] Hooks GA doc updates [DO NOT MERGE]

seejee opened the pull request:
Added sections for raising errors in the CCE and PreUserRegistration hooks. I also removed the beta tags from all of the hooks articles.

Commit:
Added section about new client credential exchange hook error.

Commit:
Add section to Pre User Registration about denying a signup.

Commit:
Remove beta tags from all hooks articles.

rkhoriander committed:
Minor copyedits

rkhoriander committed:
Minor copyedits

rkhoriander commented on articles/hooks/extensibility-points/pre-user-registration.md:
```diff
@@ -172,3 +172,28 @@ When we run this Hook, the response object is:
+
+## Sample script: Customize the error message and language for user messages
+
+In this example, we use a Hook to prevent a user from registering, then return a custom error message in our tenant logs and show a custom, translated error message to the user when they are denied. To use the translation functionality, your tenant must be configured to use the [Universal Login - New Experience](/universal-login/new).
```
@seejee  Which part of this is available only with Universal Login - New? Is it just the translation parameter, the ability to show a customized message to the user (regardless of language), or is it both?

seejee replied:
Ah good point.

Both the translation parameter AND the error message showing up in the signup form are only available in Universal Login - New.

Commit:
Clean up Hooks language in the documentation.

- Added sync/async column
- Removed some connection specific language
- Made it clear that the CCE hook is for machine to machine communication.

seejee commented on articles/hooks/extensibility-points/index.md:
```diff
@@ -13,14 +12,16 @@ useCase: extensibility-hooks
-| [Post-User Registration](/hooks/extensibility-points/post-user-registration) | `post-user-registration` | Database, Passwordless | Implement custom actions that execute asynchronously from the Auth0 authentication process after a new user registers and is added to the database. |
-| [Post-Change Password](/hooks/extensibility-points/post-change-password) | `post-change-password` | Database | Implement custom actions to be executed after a successful user password change. |
-| [Send Phone Message](/hooks/extensibility-points/send-phone-message) | `send-phone-message` | MFA | Implement a custom SMS provider to deliver MFA one-time-use codes |
+| Extensibility&nbsp;Point | Trigger ID | Connection Type(s) | Description | synchronous/asynchronous |
```
I added this column header because I couldn't think of a better word to encompass sync vs. async. Any ideas?

seejee commented on articles/hooks/extensibility-points/client-credentials-exchange.md:
```diff
@@ -16,9 +15,7 @@ v2: true
 
-At the Client Credentials Exchange extensibility point, Hooks allow custom actions to be executed after an <dfn data-key="access-token">Access Token</dfn> is issued through the Authentication API [`POST /oauth/token` endpoint](/api/authentication#client-credentials-flow) using the [Client Credentials Flow](/flows/concepts/client-credentials). For example, you may add custom claims to the Access T ...
-
-The Client Credentials Exchange extensibility point is available for [Database Connections](/connections/database) and [Passwordless Connections](/connections/passwordless).
```
Removed this line since it's not correct.

rkhoriander committed:
Minor copyedits

rkhoriander committed:
Update related extensibility points

rkhoriander committed:
Clarify necessary use of UL for pre-reg hook example

Commit:
Merge pull request #1 from auth0/pr/9052

Final updates to Hooks docs
